shopinvader_multi_cart/services/carts.py

```diff
@@ -37,41 +37,37 @@ def get(self, _id):
 
     def select(self, _id):
         """Selects the cart with the given id.
+
+        If there's already an active cart, it will be saved.
+
         :param _id: int
         :return: dict/json
         """
+        # First, save the current cart (if any)
+        self.component("cart").save()
+        # Then, select the new one
         cart = self._get(_id)
-        cart_data = self._to_json(cart)[0]
-        return {
-            "data": cart_data,
-            "store_cache": {"cart": cart_data},
-            "set_session": {"cart_id": cart.id},
-        }
+        self._restore(cart)
+        return self.component("cart")._to_json(cart)
 
     def delete(self, _id):
         """Deletes the cart with the given id.
         :param _id: int
         """
         cart = self._get(_id)
         cart.unlink()
-        if self.shopinvader_session.get("cart_id") == _id:
-            return self._clear_session_cart()
         return {}
 
     # The following methods are 'private' and should be never NEVER be called
     # from the controller.
     # All params are trusted as they have been checked before
 
-    def _clear_session_cart(self):
-        return {
-            "data": {},
-            "store_cache": {"cart": {}},
-            "set_session": {"cart_id": 0},
```

Comment:
you should probably return this to set the cart to the new cart (at least w/ Locomotive)

Reply:
AFAICS it's not necessary, however I haven't tested with locomotive so I may be wrong

On cart select (`saved` -> `cart`), this is handled here: 

https://github.com/shopinvader/odoo-shopinvader/blob/c95a7ecc2dfe1034b60e5fd4a47c47a77be3ae91/shopinvader_multi_cart/services/carts.py#L51

On cart save (`cart` -> `saved`), it's handled here:

https://github.com/shopinvader/odoo-shopinvader/blob/c95a7ecc2dfe1034b60e5fd4a47c47a77be3ae91/shopinvader_multi_cart/services/cart.py#L17


Basically the `_to_json` method will prepare the response locomotive needs:

https://github.com/shopinvader/odoo-shopinvader/blob/c95a7ecc2dfe1034b60e5fd4a47c47a77be3ae91/shopinvader/services/cart.py#L411-L424